lib/rules/no-typos.js

```diff
@@ -33,6 +42,17 @@ module.exports = {
       });
     }
 
+    function reportErrorIfLifecycleMethodCasingTypo(node) {
```

Comment:
I think we should rename the `reportErrorIfCasingTypo` function as well to contain something like `ifClassProperty`.